crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1418,6 +1429,34 @@ impl<'db> Type<'db> {
         KnownClass::NoneType.to_instance(db)
     }
 
+    /// Return the type of `tuple(sys.version_info)`.
+    ///
+    /// This is not exactly the type that `sys.version_info` has at runtime,
+    /// but it's a useful fallback for us in order to infer `Literal` types from `sys.version_info` comparisons.
+    fn version_info_tuple(db: &'db dyn Db) -> Self {
+        let target_version = Program::get(db).target_version(db);
+        let int_instance_ty = KnownClass::Int.to_instance(db);
+
+        // TODO: just grab this type from typeshed (it's a `sys._ReleaseLevel` type alias there)
+        let release_level_ty = {
+            let elements: Box<[Type<'db>]> = ["alpha", "beta", "candidate", "final"]
+                .iter()
+                .map(|level| Type::string_literal(db, level))
+                .collect();
+            Type::Union(UnionType::new(db, elements))
```

Comment:
This can be
```suggestion
            let elements = ["alpha", "beta", "candidate", "final"]
                .iter()
                .map(|level| Type::string_literal(db, level));
            UnionType::from_elements(db, elements)
```

Reply:
I think that would mean we'd do more work than we need to here. `UnionType::from_elements` goes via `UnionBuilder` to ensure that elements are properly deduplicated before the `UnionType` is constructed. But here we know just by looking at it that there are no duplicates in this union, so we can skip a layer of indirection.

https://github.com/astral-sh/ruff/blob/b3b5c191051a4ea2b81b11ed3088a09ec6d051fc/crates/red_knot_python_semantic/src/types.rs#L2714-L2726

https://github.com/astral-sh/ruff/blob/b3b5c191051a4ea2b81b11ed3088a09ec6d051fc/crates/red_knot_python_semantic/src/types/builder.rs#L107-L113

I'll add a comment to explain why I'm not using `UnionType::from_elements` here.